### ISSUE TITLE
Nethserver-cockpit is not installable

### ISSUE THREAD

**Issue by markVnl** [NethServer/arm-dev#32]
nethserver-cockpit requires nethserver-subscription and the latter requires puppet-agent which is not availible for armhfp (and probably not for aarch64).

Moreover we do not have a subscription model for arm which is a community effort.

Also See:
https://community.nethserver.org/t/nethserver-on-raspberry-pi/2851/178
https://community.nethserver.org/t/release-candidate-iso-7-7-1908-rc1/13482/26

**Comment by DavidePrincipi** [comment]
Proposal:

- define an arch-dependant subpackage for nethserver-subscription, (say nethserver-subscription-inventory). The ARM implementation is just an empty RPM.
- on the UI side, "disable" the support activation by checking a file existence from nethserver-subscription-inventory

**Comment by markVnl** [comment]
to be honest this stands not very high on my prio list..

For me any solution will do;
lets ask @federicoballarini and @mrmarkuz  too

**Comment by DavidePrincipi** [comment]
> to be honest ...

:joy: I'll take care of implementing it

**Comment by DavidePrincipi** [comment]
@markVnl if the package-split is good I can complete the implementation on the UI side. Please let me know!

**Comment by markVnl** [comment]
> @markVnl if the package-split is good I can complete the implementation on the UI side. Please let me know!

Will do so.
Have a appointment right now, first thing tomorrow morning .

**Comment by DavidePrincipi** [comment]
https://community.nethserver.org/t/teaser-nethserver-cockpit-on-arm/13543

Ok, I proceed with the UI changes

**Comment by DavidePrincipi** [comment]
...after a brief analysis I think there could be no need to change the UI now because Nethesis servers do not provide aarch64 and armhfp mirrors. 

    curl 'https://mirrorlist.nethserver.com/?repo=base&arch=x86_64&nsversion=7.7.1908&usetier=no'
    curl 'https://mirrorlist.nethserver.com/?repo=base&arch=armhfp&nsversion=7.7.1908&usetier=no'

I'll ask Alessio to write in the right places that Nethesis support is provided only for x86_64 ATM...

**Comment by DavidePrincipi** [comment]
New packages in nethserver-testing 7.7

**Comment by markVnl** [comment]
> New packages in nethserver-testing 7.7

Do I understand correctly latest PR#26 should result in:
 
![image](https://user-images.githubusercontent.com/17218846/66478091-e090e980-ea99-11e9-82f9-674b83dfbefa.png)

After an attempt to register a non_x86_64 system?

**Comment by DavidePrincipi** [comment]
Yes it is the expected output! However we could improve the messages to clarify it is not expected to work with non x86_64 arches /cc @alefattorini @gsanchietti

**Comment by DavidePrincipi** [comment]
**Test cases**

Try both on x86_64 and ARM architectures

1. the nethserver-support package from testing must upgrade existing packages by requiring/adding the new arch-dependant `nethserver-support-inventory` RPM
2. the nethserver-support-inventory RPM ships the inventory scripts and pulls in facter only for x86_64. For ARM builds the file list must be almost empty (apart from docs)
3. Check out the NS7 installation procedure based on CentOS
4. The subscription page always fail at the validation step on non x86_64 arches. It must work as usual on x86_64.

**Comment by DavidePrincipi** [comment]
News from @nethbot :

- nethserver-subscription-inventory-3.5.0-1.2.g69d7e72.ns7.x86_64.rpm to 7.7.1908/testing (x86_64)
- nethserver-subscription-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (x86_64)
- nethserver-subscription-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (armhfp)
- nethserver-subscription-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (aarch64)
- nethserver-subscription-ui-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (x86_64)
- nethserver-subscription-ui-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (armhfp)
- nethserver-subscription-ui-3.5.0-1.2.g69d7e72.ns7.noarch.rpm to 7.7.1908/testing (aarch64)

**Comment by markVnl** [comment]
Will build nethserver-support-inventory for armhfp asap (as in now) and upload it to  /testing (armhfp)

**Comment by gsanchietti** [comment]
> However we could improve the messages to clarify it is not expected to work with non x86_64 arches

I think that for now is good enough ;)

**Comment by DavidePrincipi** [comment]
@markVnl, please note I added some `sed` commands that fix the version number and the changelog of `nethserver-subscription-inventory.spec`. Those commands run automatically during the Travis build. Please remember to run them manually for the ARM release.

https://github.com/NethServer/nethserver-subscription/blob/69d7e729afe049af8c22b3abae7264fadfa3b3ab/.travis.yml#L28-L29

**Comment by markVnl** [comment]
> @markVnl, please note I added some `sed` commands that fix the version number and the changelog of `nethserver-subscription-inventory.spec`. Those commands run automatically during the Travis build. Please remember to run them manually for the ARM release.

Oops to late already uploaded to arm-testing... going to correct it

**Comment by DavidePrincipi** [comment]
No problem for the testing build! Just keep it in mind for the release to update.

**Comment by markVnl** [comment]
actually even if you do this with an manual test-build (as just done) , `make-sprm` undo(es) the sed commands in the actual build.. ;)

**EDIT:**
To summarize armhfp (=armv7hl)  nethserver-subscription-inventory package uploaded:
packages.nethserver.org/nethserver/7.7.1908/testing/armhfp/Packages/nethserver-subscription-inventory-3.5.0-1.2.g69d7e72.ns7.armv7hl.rpm 

@mrmarkuz @federicoballarini @dz00te can you test? 
As soon as a armhfp image is out will ping you on the community again

**Comment by markVnl** [comment]
> @markVnl, please note I added some `sed` commands that fix the version number and the changelog of `nethserver-subscription-inventory.spec`. Those commands run automatically during the Travis build. Please remember to run them manually for the ARM release.
> 
> https://github.com/NethServer/nethserver-subscription/blob/69d7e729afe049af8c22b3abae7264fadfa3b3ab/.travis.yml#L28-L29

For the future we may autotomize this (for reference) two commands which produce the (empty) aarch64 nethserver-subscription-inventory package on x86_64 buildhost: 

```
mock -D "dist .ns7" --resultdir=. -r nethserver-7-x86_64 nethserver-subscription-inventory-3.5.0-1.2.g69d7e72.ns7.src.rpm --arch=aarch64
rpmbuild -bb --target=aarch64 nethserver-subscription-inventory.spec

```

**Comment by markVnl** [comment]
> 
> 
> **Test cases**
> 
> Try both on x86_64 and ARM architectures
> 
>     1. the nethserver-support package from testing must upgrade existing packages by requiring/adding the new arch-dependant `nethserver-support-inventory` RPM

**n/a** for armhfp and aarch64 (initial package)
> 
>     2. the nethserver-support-inventory RPM ships the inventory scripts and pulls in facter only for x86_64. For ARM builds the file list must be almost empty (apart from docs)

**OKE** for armhfp and aarch64

> 
>     3. Check out the NS7 installation procedure based on CentOS
> 

not possible without #31

>     4. The subscription page always fail at the validation step on non x86_64 arches. It must work as usual on x86_64.

**OKE** armhfp and aarch64

**Comment by DavidePrincipi** [comment]
Released in nethserver-updates 7.6 7.7:

- nethserver-subscription-inventory-3.5.1-1.ns7.x86_64
- nethserver-subscription-ui-3.5.1-1.ns7.noarch
- nethserver-subscription-3.5.1-1.ns7.noarch

@markVnl do you want to close this issue after releasing the `nethserver-subscription-inventory` package for ARM?

**Comment by markVnl** [comment]
> do you want to close this issue after releasing the `nethserver-subscription-inventory` package for ARM?

I will upload packages late afternoon -evening-  and close this afterwards

**Comment by markVnl** [comment]
Released in nethserver-updates **7.7** armhfp & aarch64 :

- [nethserver-subscription-inventory-3.5.1-1.ns7.armv7hl.rpm](http://packages.nethserver.org/nethserver/7.7.1908/updates/armhfp/Packages/nethserver-subscription-inventory-3.5.1-1.ns7.armv7hl.rpm)
- [nethserver-subscription-inventory-3.5.1-1.ns7.aarch64.rpm]( http://packages.nethserver.org/nethserver/7.7.1908/updates/aarch64/Packages/nethserver-subscription-inventory-3.5.1-1.ns7.aarch64.rpm)
